mod/01_rating/spec/set/type_plus_right/metric/all_metric_values_spec.rb

```diff
@@ -18,103 +18,57 @@
     end
   end
 
-  describe "#get_params" do
-    it "returns value from params" do
-      Card::Env.params["offset"] = "5"
-      expect(all_values.get_params("offset", 0)).to eq(5)
-    end
-
-    it "returns default" do
-      expect(all_values.get_params("offset", 0)).to eq(0)
-    end
-  end
-
-  describe "#get_cached_values" do
-    it "returns correct cached metric values" do
-      results = all_values.get_cached_values
-      value_idx = 1
-      @companies.each do |company|
-        expect(results.key?(company.name)).to be_truthy
+  describe "format :json" do
+    describe "view :core" do
+      subject do
+        JSON.parse @metric.all_metric_values_card.format(:json).render_core
+      end
 
-        0.upto(3) do |i|
-          found_expected = results[company.name].any? do |row|
-            row[:year] == (2015 - i).to_s &&
-              row[:value] == (value_idx * 5 + i).to_s
-          end
-          expect(found_expected).to be_truthy
-        end
-        value_idx += 1
+      it "uses ids as keys" do
+        expect(subject.keys.map(&:to_i).sort).to eq @companies.map(&:id).sort
       end
-    end
-    context "delete a value" do
-      it "removes deleted cached value" do
-        Card::Auth.as_bot do
-          Card["#{@metric.name}+Apple Inc.+2015"].delete
-        end
-        results = all_values.get_cached_values
-        found_unexpected = results["Apple Inc."].any? do |row|
-          row[:year] == "2015" && row[:value] == "20"
-        end
-        expect(found_unexpected).to be_falsey
+
+      it "finds all companies with values" do
+        create_or_update! "test", type: :pointer
+        expect(subject.size).to eq 4
```

Comment:
still room for improvement. Was thinking about shared examples for metric/all_metric_values_spec.rb and company/all_metric_values_spec.rb but decided isn't worth the time